### PR TITLE
Exhaustively match on `{Statement,Terminator}Kind` during const checking

### PR DESCRIPTION
**Pull request by ecstatic-morse**
This adds a pre-monomorphization error for inline assembly in a const context as well.

r? @oli-obk

### CODE REVIEW

File: src/test/ui/consts/inline_asm.stderr [hunk 1]
```diff
@@ -0,0 +1,9 @@
+error[E0019]: constant contains unimplemented expression type
+  --> $DIR/inline_asm.rs:3:1
+   |
+LL | const _: () = unsafe { llvm_asm!("nop") };
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (oli-obk) [inline src/test/ui/consts/inline_asm.stderr:5]: I wonder why all these spans are so bad. Ah, looks like  `visit_statement` does not call `super_statement` except for `Assign`.

Comment (oli-obk) [reply src/test/ui/consts/inline_asm.stderr:5]: Please walk it for `LlvmInlineAsm`, and if you feel motivated, try out what happens when you do the same for the `IfOrMatch` arm

**Comment by ecstatic-morse** [comment]
@oli-obk seems like CI is passing with the proper spans now.

**Pull request merged**